src/arm/mach/init.c

```diff
@@ -14,6 +14,12 @@
 #include <cpuinfo/internal-api.h>
 #include <cpuinfo/log.h>
 
+#define SAFE_FREE_NULL(x) 	\
```

Comment:
This macro needs to go. cpuinfo use a the following pattern to ensure allocations properly deallocated:

```c
void some_function(args) {
  struct local_structure* some_local = NULL;

  some_local = allocate_local_structure(...);

  if (...something that may fail has failed...) {
    // log error
    goto cleanup;
  }
 
  // Commit changes to global state and return without invoking cleanup code
  global_structure /* e.g. cpuinfo_processors */ = local_structure; 
  return; // early return - skip cleanup

cleanup:
  free(local_structure);
}
```